pkg/reading/service_fetch_issue_data.go

```diff
@@ -0,0 +1,68 @@
+package reading
+
+import (
+	"errors"
+	"fmt"
+	"github.com/TurnsCoffeeIntoScripts/jira-api-issue-resource/pkg/configuration"
+	"github.com/TurnsCoffeeIntoScripts/jira-api-issue-resource/pkg/helpers"
+	"github.com/TurnsCoffeeIntoScripts/jira-api-issue-resource/pkg/http/rest"
+	"github.com/TurnsCoffeeIntoScripts/jira-api-issue-resource/pkg/service"
+	"net/http"
+)
+
+type ServiceFetchIssueData struct {
+	issueId    string
+	parentKey  string
+	statusName string
+}
+
+func (s *ServiceFetchIssueData) InitJiraAPI(params configuration.JiraAPIResourceParameters) (rest.JiraAPI, error) {
+	s.issueId = params.ActiveIssue
+
+	return service.PreInitJiraAPI(s, params, http.MethodGet)
+}
+
+func (s *ServiceFetchIssueData) GetResults() map[string]string {
+	var m = make(map[string]string)
+	m[helpers.ParentIssueKey] = s.parentKey
+	m[helpers.StatusNameKey] = s.statusName
+	return m
+}
+
+func (s *ServiceFetchIssueData) SetResultsFromPrevious(result map[string]string) {
+}
+
+func (s *ServiceFetchIssueData) GetEndpoint(url string) string {
+	return fmt.Sprintf("%s/issue/%s?expand=names", url, s.issueId)
+}
+
+func (s *ServiceFetchIssueData) CreateRequestBody() []byte {
+	return nil
+}
+
+func (s *ServiceFetchIssueData) JSONResponseObject() interface{} {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [exported method ServiceFetchIssueData.JSONResponseObject should have comment or be unexported](https://app.codacy.com/manual/TurnsCoffeeIntoScripts/jira-api-resource/pullRequest?prid=5024349)